autoload/vimteractive.vim

```diff
@@ -1,9 +1,57 @@
+let s:iconn_cw_winnr = -1
+
+function! s:show_term(bufnr)
+	split
+	execute ":b " . g:vimteractive_terminal[a:bufnr]
+	wincmd p
+endfunction
+
+function! s:check_alive(bufnr)
+	if bufexists(g:vimteractive_terminal[a:bufnr])
+		return 1
+	else
+		unlet g:vimteractive_terminal[a:bufnr]
+		unlet g:vimteractive_current_term_type[a:bufnr]
+		return 0
+	endif
+endfunction
+
+" List all running terminals
+function! vimteractive#term_list(...)
+	return filter(uniq(values(g:vimteractive_terminal)), 'bufexists(v:val)')
+endfunction
+
 " Send list of lines to the terminal buffer, surrounded with a bracketed paste
 function! vimteractive#sendlines(lines)
-	if get(g:vimteractive_bracketed_paste, g:current_term_type, 1)
-		call term_sendkeys(g:vimteractive_buffer_name,"[200~" . join(a:lines, "\n") . "[201~\n")
+	let l:bufnr = bufnr('%')
+	let l:term_buffer_name = get(g:vimteractive_terminal, l:bufnr, -1)
+	if l:term_buffer_name == -1 || !s:check_alive(l:bufnr)
```

Comment:
I really like this functionality, although this code block should probably be extracted into a separate ``function! vimteractive#autostart``, as it adds quite a lot of extra code to the sendlines command.